src/core/sys/posix/signal.d

```diff
@@ -1680,90 +1711,191 @@ else version( CRuntime_UClibc )
-            } _sigfault_t _sigfault;
+                // SIGCHLD
+                struct _sigchild_t
+                {
+                    pid_t   si_pid;    // Which child
+                    uid_t   si_uid;    // Real user ID of sending process
+                    int     si_status; // Exit value or signal
+                    clock_t si_utime;
+                    clock_t si_stime;
+                } _sigchild_t _sigchld;
+
+                // SIGILL, SIGFPE, SIGSEGV, SIGBUS
+                struct _sigfault_t
+                {
+                    void*     si_addr;  // Faulting insn/memory ref
+                } _sigfault_t _sigfault;
 
-            // SIGPOLL
-            struct _sigpoll_t
-            {
-                c_long   si_band;   // Band event for SIGPOLL
-                int      si_fd;
-            } _sigpoll_t _sigpoll;
+                // SIGPOLL
+                struct _sigpoll_t
+                {
+                    c_long si_band; // Band event for SIGPOLL;
+                    int      si_fd;
+                } _sigpoll_t _sigpoll;
 
-            // SIGSYS
-            struct _sigsys_t
+                // SIGSYS
+                struct _sigsys_t
+                {
+                    void*   _call_addr;   // Calling user insn.
+                    int     _syscall;     // Triggering system call number.
+                    uint    _arch;        // AUDIT_ARCH_* of syscall.
+                } _sigsys_t _sigsys;
+
+            } _sifields_t _sifields;
+
+        nothrow @nogc:
+            @property ref pid_t si_pid() return { return _sifields._kill.si_pid; }
+            @property ref uid_t si_uid() return { return _sifields._kill.si_uid; }
+            @property ref int si_timerid() return { return _sifields._timer.si_tid;}
+            @property ref int si_overrun() return { return _sifields._timer.si_overrun; }
+            @property ref int si_status() return { return _sifields._sigchld.si_status; }
+            @property ref clock_t si_utime() return { return _sifields._sigchld.si_utime; }
+            @property ref clock_t si_stime() return { return _sifields._sigchld.si_stime; }
+            @property ref sigval si_value() return { return _sifields._rt.si_sigval; }
+            @property ref int si_int() return { return _sifields._rt.si_sigval.sival_int; }
+            @property ref void* si_ptr() return { return _sifields._rt.si_sigval.sival_ptr; }
+            @property ref void* si_addr() return { return _sifields._sigfault.si_addr; }
+            @property ref c_long si_band() return { return _sifields._sigpoll.si_band; }
+            @property ref int  si_fd() return { return _sifields._sigpoll.si_fd; }
+            @property ref void*  si_call_addr() return { return _sifields._sigsys._call_addr; }
+            @property ref int  si_syscall() return { return _sifields._sigsys._syscall; }
+            @property ref uint  si_arch() return { return _sifields._sigsys._arch; }
```

Comment:
Can these be templates for the benefit of those who use the C bindings, but don't link to druntime?

Reply:
changed to templates, removed `return` attribute as it should be inferred for templates.